kasa/discover.py

```diff
@@ -378,20 +380,30 @@ def _get_device_instance(
                 f"Unable to read response from device: {ip}: {ex}"
             ) from ex
 
-        if (
-            discovery_result.mgt_encrypt_schm.encrypt_type == "KLAP"
-            and discovery_result.mgt_encrypt_schm.lv is None
-        ):
+        if discovery_result.mgt_encrypt_schm.encrypt_type in ("KLAP", "AES"):
             type_ = discovery_result.device_type
             device_class = None
-            if type_.upper() == "IOT.SMARTPLUGSWITCH":
-                device_class = SmartPlug
+            supported_device_types: dict[str, Type[SmartDevice]] = {
+                "SMART.TAPOPLUG": TapoPlug,
+                "SMART.KASAPLUG": TapoPlug,
+                "IOT.SMARTPLUGSWITCH": SmartPlug,
+            }
+
+            if discovery_result.device_type in supported_device_types:
+                device_class = supported_device_types[discovery_result.device_type]
```

Comment:
```suggestion
            if (device_device := supported_device_types.get(discovery_result.device_type) is None:
                _LOGGER.warning("Got unsupported %s", discovery_result.device_type)
            	raise UnsupportedDeviceException(...)
```
This way we can remove one level of nesting.